Ghidra/Processors/STM/data/languages/ST7.sinc

```diff
@@ -0,0 +1,758 @@
+@endif
+JCC: "MI"               is opcode=0x2B                              { export N; }
+JCC: "NC"               is opcode=0x24                              { local tmp = !C; export tmp; }
+JCC: "NE"               is opcode=0x26                              { local tmp = !Z; export tmp; }
+JCC: "NH"               is opcode=0x28                              { local tmp = !H; export tmp; }
+@ifdef ST70
+JCC: "NM"               is opcode=0x2C                              { local tmp = !I; export tmp; }
+@endif
+@ifdef ST72
+JCC: "NM"               is opcode=0x2C                              { local tmp = ((I1 << 1) | I0) == 0; export tmp; }
+@endif
+JCC: "PL"               is opcode=0x2A                              { local tmp = !N; export tmp; }
+JCC: "UGT"              is opcode=0x22                              { local tmp = !(C | Z); export tmp; }
+JCC: "ULE"              is opcode=0x23                              { local tmp = (C | Z); export tmp; }
+
+macro nzFlags(value) {
+  Z = value == 0;
+  N = value s< 0;
+}
+
+:ADC A, ASMEMOP is kind=0x09 ... & ASMEMOP & A {
+  local value = ASMEMOP;
+  local oldCarry = C;
+  C = carry(A, value);
+  H = carry(A << 4, value << 4);
+  value = value + A + oldCarry;
+  nzFlags(value);
+  A = value;
+}
+
+:ADD A, ASMEMOP is kind=0x0B ... & ASMEMOP & A {
+  local value = ASMEMOP;
+  C = carry(A, value);
+  H = carry(A << 4, value << 4);
+  value = value + A;
+  nzFlags(value);
+  A = value;
+}
+
+:AND A, ASMEMOP is kind=0x04 ... & ASMEMOP & A {
+  local value = ASMEMOP;
+  value = value & A;
+  nzFlags(value);
+  A = value;
+}
+
+:BCP A, ASMEMOP is kind=0x05 ... & ASMEMOP & A {
+  local value = ASMEMOP;
+  local result = value & A;
+  nzFlags(result);
+}
+
+:BRES imm8, "#"bit_position is $(NO_PRE) & bit_upper=0x01 & bit_lower=0x01 & bit_position ; imm8 {
+  local address:2 = imm8;
+  local value = *:1 address;
+  value = value & ~(1 << bit_position);
+  *:1 address = value;
+}
+
+:BRES [imm8], "#"bit_position is $(PRE_PIX) & bit_upper=0x01 & bit_lower=0x01 & bit_position ; imm8 {
+  local address:2 = imm8;
+  local target:2 = *:1 address;
+  local value:1 = *:1 target;
+  value = value & ~(1 << bit_position);
+  *:1 target = value;
+}
+
+:BSET imm8, "#"bit_position is $(NO_PRE) & bit_upper=0x01 & bit_lower=0x00 & bit_position ; imm8 {
+  local address:2 = imm8;
+  local value = *:1 address;
+  value = value | (1 << bit_position);
+  *:1 address = value;
+}
+
+:BSET [imm8], "#"bit_position is $(PRE_PIX) & bit_upper=0x01 & bit_lower=0x00 & bit_position ; imm8 {
+  local address:2 = imm8;
+  local target:2 = *:1 address;
+  local value:1 = *:1 target;
+  value = value | (1 << bit_position);
+  *:1 target = value;
+}
+
+:BTJF imm8, "#"bit_position, REL is $(NO_PRE) & bit_upper=0x00 & bit_lower=0x01 & bit_position ; imm8 ; REL {
+  local address:2 = imm8;
+  local value = *:1 address;
+  if (value & (1 << bit_position)) == (1 << bit_position)  goto inst_next;
+  PC = REL;
```

Comment:
I think you'll still need a `goto PC` for all of these `PC = REL` ... you might just `goto REL` instead